pkg/minikube/machine/cache_images.go

```diff
@@ -158,49 +159,70 @@ func needsTransfer(imgClient *client.Client, imgName string, cr cruntime.Manager
 
 // CacheAndLoadImages caches and loads images to all profiles
 func CacheAndLoadImages(images []string) error {
+	// This is the most important thing
 	if err := image.SaveToDir(images, constants.ImageCacheDir); err != nil {
-		return err
+		return errors.Wrap(err, "save to dir")
 	}
+
 	api, err := NewAPIClient()
 	if err != nil {
-		return err
+		return errors.Wrap(err, "api")
 	}
 	defer api.Close()
 	profiles, _, err := config.ListProfiles() // need to load image to all profiles
 	if err != nil {
 		return errors.Wrap(err, "list profiles")
 	}
+
+	succeeded := []string{}
+	failed := []string{}
+
 	for _, p := range profiles { // loading images to all running profiles
 		pName := p.Name // capture the loop variable
+
 		c, err := config.Load(pName)
 		if err != nil {
-			return err
+			// Non-fatal because it may race with profile deletion
+			glog.Errorf("Failed to load profile %q: %v", pName, err)
+			failed = append(failed, pName)
+			continue
 		}
+
 		for _, n := range c.Nodes {
 			m := driver.MachineName(*c, n)
+
 			status, err := Status(api, m)
 			if err != nil {
-				glog.Warningf("skipping loading cache for profile %s", pName)
 				glog.Errorf("error getting status for %s: %v", pName, err)
-				continue // try next machine
+				failed = append(failed, pName)
+				continue
 			}
+
 			if status == state.Running.String() { // the not running hosts will load on next start
 				h, err := api.Load(m)
 				if err != nil {
-					return err
+					glog.Errorf("Failed to load machine %q: %v", m, err)
+					failed = append(failed, pName)
+					continue
```

Comment:
I had seen this api.Load failing other tests as well. is there a chance to see what other test is failing on this and do the same for them ?

Reply:
This only impacts code which loads non-primary profiles. 

Looking through our code-base, this is the only example our integration tests hit -- the other is `delete --all`. If you see it elsewhere, please open an issue.